frontend/test/pytest/test_transform.py

```diff
@@ -121,44 +166,896 @@ def qfunc(data, x, weights):
+            qml.RY(angles[1], wires=1)
+            qml.CNOT(wires=[1, 0])
+            return qml.expval(qml.Z(0))
+
+        return qfunc
+
+    qnode_control = qnode_builder("default.qubit")
+    qnode_backend = qnode_builder(backend)
+
+    jax_jit = jax.jit(qnode_control)
+    compiled = qjit(qnode_backend)
+
+    params = [0.2, 0.3]
+    expected = jax_jit(params)
+    observed = compiled(params)
+    assert np.allclose(expected, observed)
+
+    _, expected_shape = jax.tree_util.tree_flatten(expected)
+    _, observed_shape = jax.tree_util.tree_flatten(observed)
+    assert expected_shape == observed_shape
+
+
+def test_undo_swaps(backend):
+    """Test undo_swaps"""
+
+    def qnode_builder(device_name):
+        """Builder"""
+
+        @qml.transforms.undo_swaps
+        @qml.qnode(qml.device(device_name, wires=3), interface="jax")
+        def qfunc():
+            qml.Hadamard(wires=0)
+            qml.X(1)
+            qml.SWAP(wires=[0, 1])
+            qml.SWAP(wires=[0, 2])
+            qml.Y(0)
+            return qml.expval(qml.Z(0))
+
+        return qfunc
+
+    qnode_control = qnode_builder("default.qubit")
+    qnode_backend = qnode_builder(backend)
+
+    jax_jit = jax.jit(qnode_control)
+    compiled = qjit(qnode_backend)
+
+    expected = jax_jit()
+    observed = compiled()
+    assert np.allclose(expected, observed)
+
+    _, expected_shape = jax.tree_util.tree_flatten(expected)
+    _, observed_shape = jax.tree_util.tree_flatten(observed)
+    assert expected_shape == observed_shape
+
+
+class TestMitigate:
+    """Test error mitigation transforms"""
+
+    @pytest.mark.xfail(reason="JIT and QJIT return different values.")
+    def test_fold_global(self, backend):
+        """Test fold_global"""
+
+        def qnode_builder(device_name):
+            """Builder"""
+
+            @partial(qml.transforms.fold_global, scale_factor=2)
+            @qml.qnode(qml.device(device_name, wires=3), interface="jax")
+            def qfunc(x):
+                qml.RX(x[0], wires=0)
+                qml.RY(x[1], wires=1)
+                qml.RZ(x[2], wires=2)
+                qml.CNOT(wires=(0, 1))
+                qml.CNOT(wires=(1, 2))
+                qml.RX(x[3], wires=0)
+                qml.RY(x[4], wires=1)
+                qml.RZ(x[5], wires=2)
+                return qml.expval(qml.Z(0) @ qml.Z(1) @ qml.Z(2))
+
+            return qfunc
+
+        qnode_control = qnode_builder("default.qubit")
+        qnode_backend = qnode_builder(backend)
+
+        jax_jit = jax.jit(qnode_control)
+        compiled = qjit(qnode_backend)
+
+        x = np.arange(6)
+        expected = jax_jit(x)
+        observed = compiled(x)
+        assert np.allclose(expected, observed)
+
+        _, expected_shape = jax.tree_util.tree_flatten(expected)
+        _, observed_shape = jax.tree_util.tree_flatten(observed)
+        assert expected_shape == observed_shape
+
+    @pytest.mark.xfail(reason="JIT and QJIT return different values.")
```

Comment:
How about qjit and plain pennylane?

Reply:
They also give different values. I will open a bug report.